net-dns/dnsmasq/dnsmasq-2.80-r1.ebuild

```diff
@@ -0,0 +1,213 @@
+# Copyright 1999-2019 Gentoo Authors
+# Copyright 2017-2018 Sony Interactive Entertainment Inc.
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit toolchain-funcs flag-o-matic user systemd
+
+DESCRIPTION="Small forwarding DNS server"
+HOMEPAGE="http://www.thekelleys.org.uk/dnsmasq/doc.html"
+SRC_URI="http://www.thekelleys.org.uk/dnsmasq/${P}.tar.xz"
+
+LICENSE="|| ( GPL-2 GPL-3 )"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~mips ~ppc ~ppc64 ~s390 ~sh ~sparc ~x86 ~x86-fbsd"
+
+IUSE="auth-dns conntrack dbus +dhcp dhcp-tools dnssec +dumpfile id idn libidn2"
+IUSE+=" +inotify ipv6 lua nls script selinux static tftp"
+
+PATCHES=(
+	"${FILESDIR}/${P}-nettle-3.5.patch"
+)
+
+DM_LINGUAS=(de es fi fr id it no pl pt_BR ro)
+
+BDEPEND="app-arch/xz-utils
+	virtual/pkgconfig
+	nls? ( sys-devel/gettext )
+"
+
+COMMON_DEPEND="dbus? ( sys-apps/dbus:= )
+	idn? (
+		!libidn2? ( net-dns/libidn:0= )
+		libidn2? ( >=net-dns/libidn2-2.0:= )
+	)
+	lua? ( dev-lang/lua:* )
+	conntrack? ( net-libs/libnetfilter_conntrack:= )
+	nls? ( sys-devel/gettext )
+"
+
+DEPEND="${COMMON_DEPEND}
+	dnssec? (
+		dev-libs/nettle:=[gmp]
+		static? ( >=dev-libs/nettle-3.4[static-libs(+)] )
+	)
+"
+
+RDEPEND="${COMMON_DEPEND}
+	dnssec? (
+		!static? (
+			>=dev-libs/nettle-3.4:=[gmp]
+		)
+	)
+	selinux? ( sec-policy/selinux-dnsmasq )
+"
+
+REQUIRED_USE="dhcp-tools? ( dhcp )
+	lua? ( script )
+	libidn2? ( idn )"
+
+use_have() {
+	local useflag no_only uword
+	if [[ ${1} == '-n' ]]; then
+		no_only=1
+		shift
+	fi
+	useflag="${1}"
+	shift
+
+	uword="${1:-${useflag}}"
+	shift
+
+	while [[ ${uword} ]]; do
+		uword="${uword^^}"
+
+		if ! use "${useflag}"; then
```

Comment:
suggestion: what about `usex` instead of printf's? Maybe a bit more readable e.g.
```--- dnsmasq-2.80.ebuild	2019-05-30 11:00:56.436119428 +0200
+++ dnsmasq-2.80-r1.ebuild	2019-06-28 16:56:40.765086805 +0200
@@ -55,7 +55,8 @@
 	libidn2? ( idn )"
 
 use_have() {
-	local useflag no_only uword
+	local useflag no_only uword define_have 
+
 	if [[ ${1} == '-n' ]]; then
 		no_only=1
 		shift
@@ -68,12 +69,9 @@
 
 	while [[ ${uword} ]]; do
 		uword="${uword^^}"
+		[[ $no_only = 1 ]] || define_have="-DHAVE_${uword}"
+		usex "${useflag}" "$define_have" "-DNO_${uword}"
 
-		if ! use "${useflag}"; then
-			printf -- " -DNO_%s" "${uword}"
-		elif [[ -z "${no_only}" ]]; then
-			printf -- " -DHAVE_%s" "${uword}"
-		fi
 		uword="${1}"
 		shift
 	done
@@ -129,7 +127,7 @@
 		LDFLAGS="${LDFLAGS}" \
 		COPTS="${COPTS[*]}" \
 		CONFFILE="/etc/${PN}.conf" \
-		all$(use nls && printf -- "-i18n\n")
+		all$(usex nls "-i18n" "")
 
 	use dhcp-tools && emake -C contrib/lease-tools \
 		PREFIX=/usr \
@@ -148,7 +146,7 @@
 		MANDIR=/usr/share/man \
 		COPTS="${COPTS[*]}" \
 		DESTDIR="${ED}" \
-		install$(use nls && printf -- "-i18n\n")
+		install$(usex nls "-i18n" "")
 
 	for lingua in "${DM_LINGUAS[@]}"; do
 		has ${lingua} ${LINGUAS-${lingua}} \
```